src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  RefObject, useEffect, useLayoutEffect, useMemo, useRef, useState,
+} from 'react';
+
+const images = [
+  {
+    idx: 0,
+    src: './_new/img/banner-accessories.png',
+    alt: 'Accessories',
+  },
+  {
+    idx: 1,
+    src: './_new/img/banner-phones.png',
+    alt: 'Phones',
+  },
+  {
+    idx: 2,
+    src: './_new/img/banner-tablets.png',
+    alt: 'Tablets',
+  },
+];
+
+export const Carousel = () => {
+  const container = useRef() as RefObject<HTMLDivElement>;
+  const [currTransitionX, setCurrTransitionX] = useState(0);
+  const [transitionDuration, setTransitionDuration] = useState(0);
+  const [containerWidth, setContainerWidth] = useState(0);
+  const [currImg, setCurrImg] = useState(1);
+  const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+
+  const slides = useMemo(() => {
+    const items = images.map(img => (
+      <li key={img.idx} className="carousel__item">
+        <img src={img.src} alt={img.alt} className="carousel__img" />
+      </li>
+    ));
+
+    return [
+      <li key={images.length} className="carousel__item">
+        <img
+          src={images[images.length - 1].src}
+          alt={images[images.length - 1].alt}
+          className="carousel__img"
+        />
+      </li>,
+      ...items,
+      <li key={images.length + 1} className="carousel__item">
+        <img
+          src={images[0].src}
+          alt={images[0].alt}
+          className="carousel__img"
+        />
+      </li>,
+    ];
+  }, [...images]);
```

Comment:
But as images are constant - your dependecies should be []
```suggestion
  }, []);
```